encoder-otel-brave/src/main/java/zipkin2/reporter/otel/brave/SpanTranslator.java

```diff
@@ -0,0 +1,215 @@
+  static final String PEER_SERVICE = "peer.service";
+
+  private static final Map<String, String> RENAMED_LABELS;
+
+  static {
+    RENAMED_LABELS = new LinkedHashMap<>();
+    // "http.host" is not defined in HttpTags, but is a well-known tag.
+    RENAMED_LABELS.put("http.host", ServerAttributes.SERVER_ADDRESS.getKey());
+    RENAMED_LABELS.put(HttpTags.METHOD.key(), HttpAttributes.HTTP_REQUEST_METHOD.getKey());
+    RENAMED_LABELS.put(HttpTags.PATH.key(), UrlAttributes.URL_PATH.getKey());
+    RENAMED_LABELS.put(HttpTags.ROUTE.key(), HttpAttributes.HTTP_ROUTE.getKey());
+    RENAMED_LABELS.put(HttpTags.URL.key(), UrlAttributes.URL_FULL.getKey());
+    RENAMED_LABELS.put(HttpTags.STATUS_CODE.key(), HttpAttributes.HTTP_RESPONSE_STATUS_CODE.getKey());
+  }
+
+  private final Consumer consumer;
+
+  SpanTranslator(Tag<Throwable> errorTag) {
+    this.consumer = new Consumer(errorTag, RENAMED_LABELS);
+  }
+
+  /**
+   * Converts a Zipkin Span into a OpenTelemetry Span.
+   *
+   * <p>Ex.
+   *
+   * <pre>{@code
+   * tracesData = SpanTranslator.translate(braveSpan);
+   * }</pre>
+   *
+   * @param braveSpan The Zipkin Span.
+   * @return A OpenTelemetry Span.
+   */
+  TracesData translate(MutableSpan braveSpan) {
+    TracesData.Builder tracesDataBuilder = TracesData.newBuilder();
+    Builder resourceSpansBuilder = ResourceSpans.newBuilder();
+    ScopeSpans.Builder scopeSpanBuilder = ScopeSpans.newBuilder();
+    Span.Builder spanBuilder = builderForSingleSpan(braveSpan, resourceSpansBuilder);
+    scopeSpanBuilder.addSpans(spanBuilder
+        .build());
+    InstrumentationScope.Builder scopeBuilder = InstrumentationScope.newBuilder();
+    scopeBuilder.setName(BraveScope.getName());
+    scopeBuilder.setVersion(BraveScope.getVersion());
+    scopeSpanBuilder.setScope(scopeBuilder.build());
+    resourceSpansBuilder.addScopeSpans(scopeSpanBuilder
+        .build());
+    tracesDataBuilder.addResourceSpans(resourceSpansBuilder.build());
+    return tracesDataBuilder.build();
+  }
+
+  static KeyValue stringAttribute(String key, String value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setStringValue(value))
+        .build();
+  }
+
+  static KeyValue intAttribute(String key, int value) {
+    return KeyValue.newBuilder()
+        .setKey(key)
+        .setValue(AnyValue.newBuilder().setIntValue(value))
+        .build();
+  }
+
+  private Span.Builder builderForSingleSpan(MutableSpan span, Builder resourceSpansBuilder) {
+    Span.Builder spanBuilder = Span.newBuilder()
+        .setTraceId(ByteString.fromHex(span.traceId() != null ? span.traceId() : io.opentelemetry.api.trace.SpanContext.getInvalid().getTraceId()))
+        .setSpanId(ByteString.fromHex(span.id() != null ? span.id() : io.opentelemetry.api.trace.SpanContext.getInvalid().getSpanId()))
+        .setName((span.name() == null || span.name().isEmpty()) ? "unknown" : span.name());
```

Comment:
The spec states that a non-empty string should be set. If nothing is actually set, it seems to be set to an empty string.
Here, instead of setting "unknown", I've changed it to leave it unset.

https://github.com/openzipkin-contrib/zipkin-otel/pull/9/commits/7528ba837d85fe6048af2c97e22725299545f59b